protocols/luci-proto-wireguard/htdocs/luci-static/resources/protocol/wireguard.js

```diff
@@ -55,6 +61,18 @@ return network.registerProtocol('wireguard', {
 		o.validate = validateBase64;
 		o.rmempty = false;
 
+		o = s.taboption('general', form.Button, 'generate_key', _('Generate Key'), _(''));
+		o.inputstyle = 'apply';
+		o.onclick = L.bind(function(ev) {
+			return  generateKey().then(L.bind(function(result) {
+				var keyInput = document.getElementById("widget.cbid.network." + s.section + ".private_key" );
+				keyInput.value = result.keys.priv;
```

Comment:
... and changing this to `result.priv || ''` (the `expect` above will take care of only returning the sub-object). You should also add the `|| ''` to avoid setting the input value to `null` in case the call failed.

Reply:
Thanks for the tip. Changed.